src/components/3d_editor/advanced_geometry/python_transformation/PythonTransformation.tsx

```diff
@@ -0,0 +1,389 @@
+import Dialog from "@exabyte-io/cove.js/dist/mui/components/dialog/Dialog";
+import IconByName from "@exabyte-io/cove.js/dist/mui/components/icon/IconByName";
+import PyodideLoader from "@exabyte-io/cove.js/dist/other/pyodide";
+import theme from "@exabyte-io/cove.js/dist/theme";
+import { Made } from "@exabyte-io/made.js";
+import Button from "@mui/material/Button";
+import DialogContent from "@mui/material/DialogContent";
+import Grid from "@mui/material/Grid";
+import Paper from "@mui/material/Paper";
+import Stack from "@mui/material/Stack";
+import Typography from "@mui/material/Typography";
+import React from "react";
+import NPMsAlert from "react-s-alert";
+
+import CodeExecutionControls, { ExecutionStatus } from "./CodeExecutionControls";
+import ExecutionCell, { ExecutionCellState } from "./ExecutionCell";
+import MaterialsSelector from "./MaterialsSelector";
+import TransformationSelector, { Transformation } from "./TransformationSelector";
+
+interface PythonTransformationProps {
+    materials: Made.Material[];
+    show: boolean;
+    onSubmit: (newMaterials: Made.Material[]) => void;
+    onHide: () => void;
+}
+
+interface PythonTransformationState {
+    materials: Made.Material[];
+    selectedMaterials: Made.Material[];
+    newMaterials: Made.Material[];
+    executionStatus: ExecutionStatus;
+    // TODO: import type for Pyodide when they are available in Cove.js
+    // @ts-ignore
+    pyodide: any;
+    transformation: Transformation | null;
+    pythonCode: string;
+    pythonOutput: string;
+    executionCells: ExecutionCellState[];
+}
+
+interface PyodideDataMap {
+    [key: string]: string | PyodideDataMap;
+}
+
+// TODO: merge SOF-7138 into dev and update the URL below
+const GITHUB_API_URL =
```

Comment:
Explain why you are using API and maybe put a docs link, explain what this returns.

```
{
name: "README.md",
path: "other/python_transformations/README.md",
sha: "841f5bfd45b5da221d02e4fafc6814c3f868295d",
size: 677,
url: "https://api.github.com/repos/Exabyte-io/api-examples/contents/other/python_transformations/README.md?ref=dev",
html_url: "https://github.com/Exabyte-io/api-examples/blob/dev/other/python_transformations/README.md",
git_url: "https://api.github.com/repos/Exabyte-io/api-examples/git/blobs/841f5bfd45b5da221d02e4fafc6814c3f868295d",
download_url: "https://raw.githubusercontent.com/Exabyte-io/api-examples/dev/other/python_transformations/README.md",
type: "file",
_links: {
self: "https://api.github.com/repos/Exabyte-io/api-examples/contents/other/python_transformations/README.md?ref=dev",
git: "https://api.github.com/repos/Exabyte-io/api-examples/git/blobs/841f5bfd45b5da221d02e4fafc6814c3f868295d",
html: "https://github.com/Exabyte-io/api-examples/blob/dev/other/python_transformations/README.md"
}
}
```

Call this `FILE_CONFIGS_FETCH_URL`